filelog.go

```diff
@@ -187,12 +189,33 @@ func (w *FileLogWriter) intRotate() error {
 	w.daily_opendate = now.Day()
 
 	// initialize rotation values
-	w.maxlines_curlines = 0
-	w.maxsize_cursize = 0
+	lineCount, _ := lineCounter(w.file)
```

Comment:
I'm not convinced this actually works, given that we appended to the file on line 186 and the file pointer is now at the end.

Reply:
Ah good point, I had it up higher when I did my original testing and moved it down here so it was closer to what was using it. Let me re-test and fix it if necessary

Reply:
Ok, I'm not crazy and it does work in it's current position. Added a unit test for sanity. Not exactly sure why the write isn't moving the file cursor to the end..

Reply:
Hmmm, looks like it's because `FormatLogRecord(w.header, &LogRecord{Created: now})` is just returning blank space. Not sure what the point of that is

Reply:
Yeah, in my testing if `fmt.Printf` doesn't actually write anything (an empty string, not a blank space), it looks like the file pointer stays at the beginning, allowing this read. But if anyone uses `SetHeadFoot` and actually configures a header, I think this will break.

I'm also a bit worried about invoking `Size()` on the file pointer before we've necessarily synced the header write to disk. I realize it's almost certainly not going to matter (I can't imagine the header being big), but I'm wondering if this could all be simplified if we opened the file for reading/stating first, and then reopened for appending, vs. trying to juggle the two pointers.

Reply:
Yeah I can do that just to be safer and cleaner